app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -69,8 +69,8 @@ public void onClick(View v) {
             @Override
             public void onClick(View view) {
                 animation.start();
-                MainActivity.darkMode = !MainActivity.darkMode;
-                UIHelpers.lightDark(v, MainActivity.darkMode);
+                UIHelpers.darkMode = !UIHelpers.darkMode;
```

Comment:
These two lines appear multiple times throughout your code atm, this could be simplified by adding another method to UIHelpers such as toggleDarkMode, this can be helpful for any future occurrences where you have to change parts of this code segment and you won't have to do so a multitude of times like you had to do here by changing all of your references to MainActivity with UIHelpers